source/templates/displaying-the-keys-in-an-object.md

```diff
@@ -0,0 +1,131 @@
+If you need to display all of the keys or values of a
+JavaScript object in your template, you can use the `{{#each-in}}`
+helper:
+
+```/app/components/store-categories.js
+import Component from "ember-component";
+
+export Component.extend({
+  willRender() {
+    // Set the "categories" property to a JavaScript object
+    // with the category name as the key and the value a list
+    // of products.
+    this.set('categories', {
+      "Bourbons": ["Bulleit", "Four Roses", "Woodford Reserve"],
+      "Ryes": ["WhistlePig", "High West"]
+    });
+  }
+});
+```
+
+```/app/templates/components/store-categories.hbs
+<ul>
+  {{#each-in categories as |category, products|}}
+    <li>{{category}}
+      <ol>
+        {{#each products key="@item" as |product|}}
+          <li>{{product}}</li>
+        {{/each}}
+      </ol>
+    </li>
+  {{/each}}
+</ul>
+```
+
+The template inside of the `{{#each-in}}` block is repeated once for
+each key in the passed object. The first block parameter (`category` in
+the above example) is the key for this iteration, while the second block
+parameter (`product`) is the actual value of that key.
+
+The above example will print a list like this:
+
+```html
+<ul>
+  <li>Bourbons
+    <ol>
+      <li>Bulleit</li>
+      <li>Four Roses</li>
+      <li>Woodford Reserve</li>
+    </ol>
+  </li>
+  <li>Ryes
+    <ol>
+      <li>WhistlePig</li>
+      <li>High West</li>
+    </ol>
+  </li>
+</ul>
+```
+
+### A Note About Automatic Updates
+
+Note that the `{{#each-in}}` helper does **not** observe property
```

Comment:
I think the concept that a rerender is needed when you add props is important information IMHO. Can we remove the 'Note that' lead in from this sentence?